Upgrade to terraform sdkv2 2.35

mdelaossa opened the pull request:
Also took the opportunity to move a bit of code from the legacy client to the new client

As part of the TF 2.35 upgrade, a test had to change due to stricter data handling by TF

Tested with
![image](https://github.com/user-attachments/assets/f8ff042f-b473-4f60-a5b9-cae0c3e8e0b3)

aaw commented on aptible/resource_app.go:
```diff
@@ -641,72 +641,91 @@ func scaleServices(d *schema.ResourceData, meta interface{}) error {
+
+			payload := aptibleapi.NewCreateOperationRequest("scale")
+			if oldServiceData["container_count"] != containerCount && containerCount != -1 {
+				log.Println("Updating containers-count")
```
You could make this log line more useful for someone in the future by adding a from/to, e.g.:

`log.Printf("Updating container count from %v to %v", oldServiceData["container_count"], containerCount)`

mdelaossa replied:
Removed, this was debug output while I was testing the original reason why I was touching the provider

aaw commented on aptible/resource_app.go:
```diff
@@ -641,72 +641,91 @@ func scaleServices(d *schema.ResourceData, meta interface{}) error {
+				payload.SetContainerCount(int32(containerCount.(int)))
 			}
-			err = client.ScaleService(service.ID, containerCount, memoryLimit, containerProfile)
+			log.Printf("memory limit old: %s, new: %s", oldServiceData["container_memory_limit"], memoryLimit)
```
Probably should just remove this log statement in favor of a more informative message on 695, something like:

`log.Printf("Updating memory limit from %v to %v", oldServiceData["container_memory_limit"], memoryLimit)`

mdelaossa replied:
Removed, this was debug output while I was testing the original reason why I was touching the provider. All of these conditionals were actually for that failed thing, and I forgot to replace. Will undo the conditionals and run another set of integration tests

aaw commented on aptible/resource_app.go:
```diff
@@ -641,72 +641,91 @@ func scaleServices(d *schema.ResourceData, meta interface{}) error {
 			}
+
+			_, err = legacy.WaitForOperation(int64(resp.Id))
+			if err != nil {
```
You could just return `err` here and it would do the same thing as lines 708-710

aaw commented on aptible/resource_app.go:
```diff
@@ -322,7 +322,7 @@ func resourceAppCreate(d *schema.ResourceData, meta interface{}) error {
 	// at the time of deployment.
 	// TODO: We can check for services scaled to 1 GB/1 container before scaling.
-	err = scaleServices(d, meta)
+	err = scaleServices(context.Background(), d, meta)
```
Might be an improvement to get an API context at the beginning of this method and pass that to scaleServices and updateServices instead of passing `context.Background()` to both and then converting it. Then again, I don't know that we're actually using the context for anything fancy so it might not matter, so feel free to ignore this comment if so.

mdelaossa commented on aptible/resource_app.go:
```diff
@@ -641,72 +641,91 @@ func scaleServices(d *schema.ResourceData, meta interface{}) error {
-		processType := serviceData["process_type"].(string)
-
+		// https://stackoverflow.com/a/74383278
+		service := s
```
ooh that's a nice change!

mdelaossa commented:
Thanks @aaw, I believe I addressed all your comments. Also, here's the new run of the integration tests I had promised
![image](https://github.com/user-attachments/assets/edf759aa-2520-4a49-b526-ea2d36fba943)
